main.go

```diff
@@ -223,19 +300,42 @@ func main() {
 		os.Exit(0)
 	}
 
-	instance_id, _ := uuid4()
+	ctx := context.Background()
+	useConfig := *configFile != ""
```

Comment:
I don't think so. Even though `flag.String` returns a pointer to a string the default value will be set if nothing is provided, so it will never be `nil` as far as I can tell.